homeassistant/components/blueprint/models.py

```diff
@@ -96,7 +96,14 @@ def name(self) -> str:
     @property
     def inputs(self) -> dict[str, Any]:
         """Return blueprint inputs."""
-        return self.data[CONF_BLUEPRINT][CONF_INPUT]  # type: ignore[no-any-return]
+        inputs = {}
+        for key, value in self.data[CONF_BLUEPRINT][CONF_INPUT].items():
+            if value and CONF_INPUT in value:
+                for key, value in value[CONF_INPUT].items():
+                    inputs[key] = value
+            else:
+                inputs[key] = value
+        return inputs
```

Comment:
The requests was to do the flattening in frontend, now the flattening happens in core instead if I understand it correctly. Will this have any side effect on existing blueprinted automations?

Reply:
Flattening in frontend is what @bramkragten meant here: https://github.com/home-assistant/core/pull/110513#issuecomment-2018148737 unless I misunderstand.
However, the solution you've come with seems fine too, I just want another pair of eyes on it.

Reply:
Never mind, I misunderstood what's happening here.